### ISSUE TITLE
List of style checks?

### ISSUE THREAD

**Issue by sebastiansauer** [NMarkgraf/RmdStyleChecker#1]
Hi @NMarkgraf,

great idea. Could you please provide a list/docu file of the checks that are being conducted? Those unlucky beings who are unfluent in Python will reside in thankfulness 😄

**Comment by NMarkgraf** [comment]
You are right! - This definitely needs a written style guide, so anybody will know what is checked by this tool.

I put that on my list. - But currently only blank lines before "#" und "##" are checked an unnecessary whitespaces at the end of a line.

Nothing is changes in the document, so this is more or less just a tool that warns you.

Also I know, there should be unit tests and some examples. - But, currently the weather is fine and the garden need some work and ... 

If you like to try it simply:

1. On *nix like systems do `chmod a+x rmdstlyechecker.py`
    first, which will make it an executable.
    
    On w*ndows systems you need to add always a`python ` or a `python3 ` before `rmdstylechecker.py`!

2. Simply run it with
```
rmdstylechecker.py RmarkdownFileYouWantToCheck.Rmd
```
or 
```
python3 rmdstylechecker.py RmarkdownFileYouWantToCheck.Rmd
```

If you get nothing but the the number of lines it has scanned, everything is all right. Else you get the line number and the problem found.

Pretty easy!

Yours
Norman

PS: And yes! - This could be written in R directly, but I currently prefer Python...

**Comment by sebastiansauer** [comment]
Thanks! I'll check that an get back to you.

**Comment by NMarkgraf** [comment]
Thanks a lot! --- I've started to write some of my thought (in german) into my [blog](http://sefiroth.net/nab/post/auch-rmarkdown-dateien-sollten-sich-regeln-halten/). If anybody has any ideas what might get on this list, please drop me a line!
